pkg/controller/operatingsystemconfig/generator/generator.go

```diff
@@ -68,18 +67,10 @@ func additionalValues(osc *extensionsv1alpha1.OperatingSystemConfig) (map[string
 	return values, nil
 }
 
-// isSupportedOscType checks if the OperatingSystemConfig's type is one of those for which this extension is responsbile
-func isSupportedOscType(osc *generator.OperatingSystemConfig) bool {
-	return osc.Object.Spec.Type == gardenlinux.OSTypeGardenLinux ||
-		osc.Object.Spec.Type == memoryone.OSTypeMemoryOneGardenLinux
-}
-
 // Generate generates a Garden Linux specific cloud-init script from the given OperatingSystemConfig.
 func (g *GardenLinuxCloudInitGenerator) Generate(logger logr.Logger, osc *generator.OperatingSystemConfig) ([]byte, *string, error) {
```

Comment:
Maybe I am not fully understanding the flow and/or intention but in https://github.com/gardener/gardener-extension-os-gardenlinux/blob/e9ab9eeda3623b13715959f4b103bd1af0c9041f/pkg/controller/operatingsystemconfig/actuator.go#L88, the data for CloudInit is generated by the `handleProvisionOSC()` function, yet here, if `useGardenerNodeAgent` is not used, the old template based generator is used. From what I can see, the actual output seems to be the same for both generating paths.
Can't we do away with the templater and use `handleProvisionOSC()` and `handleReconcileOSC()` exclusively for both cases?

Reply:
Maybe we could, but I would prefer to keep things separate for now so that the old legacy generator can be easily deleted later when `gardener-node-agent` is GA and the `cloud-config-downloader` scenario fully disappeared. This way, we don't change anything for the current world (no `gardener-node-agent` in place), but just for the new feature when the feature gate is turned on. Hence, if anything goes wrong, it's just for the new scenario, not for the old one.

We still need to run the generator anyways for the migration scenario (when the `UseGardenerNodeAgent` feature gate is turned on for the first time) where `cloud-config-downloader` still runs on existing nodes. It needs to download the new user data (containing the `gardener-node-agent` systemd unit), so we wouldn't gain much from trying to harmonize anything now here.